internal/mailer/mailer.go

```diff
@@ -27,8 +28,46 @@ type EmailProvider interface {
 
 type GetAllEmailStatusSpec struct {
 	ID           string    `db:"id" json:"id"`
-	EmailTo      string    `db:"email_to" json:"email_to"`
-	Status       string    `db:"status" json:"status"`
-	ModifiedDate time.Time `db:"modified_date" json:"modified_date"`
+	EmailTo      string    `db:"email_to"`
+	Status       string    `db:"status"`
+	ModifiedDate time.Time `db:"modified_date"`
 	database.PaginationSpec
 }
+
+type EmailStatusEnum int64
+
+const (
+	Success EmailStatusEnum = iota
+	Failed
+	InProgress
+	Completed
+)
```

Comment:
Any reason as to why we don't use `type EmailStatusEnum string` here?
it would save us a lot of time, since we don't have to keep calling the `.String()` function

Reply:
Tbh I followed the example from internet on how to implement enum in golang. They all made it like that, for example https://www.sohamkamani.com/golang/enums/ and https://stackoverflow.com/questions/41480543/how-to-make-go-print-enum-fields-as-string

Reply:
Because technically it's not enum 😂
you could do custom type with constants that looks like this

```
type EmailStatus string

const (
	EmailStatusSuccess EmailStatus = "success"
	EmailStatusFailed    EmailStatus = "failed"
)
```

Reply:
In Go, enums can be implemented using different approaches depending on the intended use case, and the two common methods are using integers (with iota) or strings. Both approaches have their own strengths and weaknesses, and choosing the right one depends on the context in which the enum will be used.

The first approach involves defining an enum as an integer type, typically with iota to automatically assign sequential values. This method is efficient in terms of memory usage and performance, as integers are lightweight and fast to compare or process. It works well for internal logic, such as conditional checks or switch statements, making the code more concise and readable. However, when debugging or logging, the enum will display its numeric value which can be less meaningful without explicit mappings to their string equivalents. Additionally, when serializing data to formats like JSON or transferring it over APIs, extra logic is required to convert these integer values into readable strings, making this approach more suitable for internal systems where the values are not exposed directly to users.

The second approach involves defining the enum as a string type, which uses human-readable labels such as "success" or "failed". This approach is ideal for scenarios where the enum values need to be serialized, logged, or presented to users. Strings are more meaningful and easier to understand in these contexts. For example, when working with JSON APIs, strings like "success" or "failed" are more intuitive than integers like 0 or 1, requiring no additional conversion. However, strings consume more memory and have slightly slower comparison performance than integers, though this difference is negligible in most applications. This makes string-based enums a great choice for applications that prioritize readability and ease of debugging over performance.